packages/server/lib/project-base.ts

```diff
@@ -28,46 +21,82 @@ import { ServerE2E } from './server-e2e'
 import system from './util/system'
 import user from './user'
 import { ensureProp } from './util/class-helpers'
-import { escapeFilenameInUrl } from './util/escape_filename'
 import { fs } from './util/fs'
-import keys from './util/keys'
 import settings from './util/settings'
 import plugins from './plugins'
 import specsUtil from './util/specs'
 import Watchers from './watchers'
 import devServer from './plugins/dev-server'
 import preprocessor from './plugins/preprocessor'
-import { SpecsStore } from './specs-store'
+import { RunnerType, SpecsStore } from './specs-store'
 import { createRoutes as createE2ERoutes } from './routes'
 import { createRoutes as createCTRoutes } from '@packages/server-ct/src/routes-ct'
-
-interface OpenOptions {
-  onOpen: (cfg: any) => Bluebird<any>
+import { checkSupportFile } from './project_utils'
+
+// Cannot just use RuntimeConfigOptions as is because some types are not complete.
```

Comment:
I rather like this way of describing locally used props.
Maybe they should not be on one line, but I like not having the whole config object available.

Reply:
I agree.

Putting them on separate lines is an option, problem is it'll be around 30-40 lines - not sure how useful this is. Ideally our linter should catch this - it doesn't seem to care, so I guess it's fine.